lms/djangoapps/teams/api_urls.py

```diff
@@ -14,7 +14,7 @@
 
 TEAM_ID_PATTERN = r'(?P<team_id>[a-z\d_-]+)'
```

Comment:
Basically, staff members can add the discussion topics through "Course Advance Settings" and they will have to give `topic_id`s for the topics they are adding so that's why there is constraint on `topic_id` while `team_id` will be automatically generated when staff member add a team to an existing topic. `team_id` cannot be altered by staff and it is following the right pattern by default. 
See [Creating Team](http://edx.readthedocs.org/projects/edx-partner-course-staff/en/latest/course_features/teams/teams_setup.html#create-a-team)